lib/ocpp/v201/charge_point.cpp

```diff
@@ -2783,31 +2783,22 @@ void ChargePoint::handle_customer_information_req(Call<CustomerInformationReques
 void ChargePoint::handle_data_transfer_req(Call<DataTransferRequest> call) {
     const auto msg = call.msg;
     DataTransferResponse response;
-    const auto vendor_id = msg.vendorId.get();
-    const auto message_id = msg.messageId.value_or(CiString<50>()).get();
-    {
-        std::lock_guard<std::mutex> lock(data_transfer_callbacks_mutex);
-        if (this->data_transfer_callbacks.count(vendor_id) == 0) {
-            response.status = ocpp::v201::DataTransferStatusEnum::UnknownVendorId;
-        } else if (this->data_transfer_callbacks.count(vendor_id) and
-                   this->data_transfer_callbacks[vendor_id].count(message_id) == 0) {
-            response.status = ocpp::v201::DataTransferStatusEnum::UnknownMessageId;
-        } else {
-            // there is a callback registered for this vendorId and messageId
-            response = this->data_transfer_callbacks[vendor_id][message_id](msg.data);
-        }
+
+    if (this->callbacks.data_transfer_callback.has_value()) {
+        response = this->callbacks.data_transfer_callback.value()(call.msg);
     }
```

Comment:
What should the response be in case the `data_transfer_callback` is empty? Currently it seems you'd just return a default-constructed response, but I'm not sure this is the right way to do that...

Reply:
UnknownVendorId seems appropriate here